core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -315,11 +316,16 @@ sealed trait Property[T] extends Element { self =>
     requireIsHardware(this)
     requireVisible()
     topBindingOpt match {
+      case Some(PropExprBinding(sourceInfo, _, op, lhs, rhs)) =>
+        // For Property expressions, we create a special kind of Arg that builds in-memory trees without creating nodes.
+        ir.PropExpr(sourceInfo, tpe.getPropertyType(), op, lhs, rhs)
```

Comment:
I don't like getting this from the Binding rather than just setting the ref. The current system is a mess but this would just add yet-another-way information is stored and later derived.